test/integration/isolated/ingress_test.go

```diff
@@ -196,3 +200,155 @@ func TestIngressGRPC(t *testing.T) {
+			cleaner := GetFromCtxForT[*clusters.Cleaner](ctx, t)
+			cluster := GetClusterFromCtx(ctx)
+			namespace := GetNamespaceForT(ctx, t)
+
+			t.Log("configuring Deployment")
+			container := generators.NewContainer("httpbin", test.HTTPBinImage, test.HTTPBinPort)
+			deployment := generators.NewDeploymentForContainer(container)
+			deployment, err := cluster.Client().AppsV1().Deployments(namespace).Create(ctx, deployment, metav1.CreateOptions{})
+			assert.NoError(t, err)
+			cleaner.Add(deployment)
+
+			t.Log("configuring Service for Deployment")
+			service := generators.NewServiceForDeployment(deployment, corev1.ServiceTypeClusterIP)
+			service, err = cluster.Client().CoreV1().Services(namespace).Create(ctx, service, metav1.CreateOptions{})
+			require.NoError(t, err)
+			cleaner.Add(service)
+
+			t.Log("configuring KongServiceFacades for Service")
+			ingressClass := GetIngressClassFromCtx(ctx)
+			serviceFacades := []*incubatorv1alpha1.KongServiceFacade{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "httpbin-facade-jpeg",
+						Annotations: map[string]string{
+							annotations.IngressClassKey: ingressClass,
+							// We'll use this annotation to test that modifications are allowed on KongServiceFacade
+							// level. This should make Kong Service created from this KongServiceFacade to always
+							// return a JPEG image.
+							annotations.AnnotationPrefix + annotations.PathKey: "/image/jpeg",
+						},
+					},
+					Spec: incubatorv1alpha1.KongServiceFacadeSpec{
+						Backend: incubatorv1alpha1.KongServiceFacadeBackend{
+							Name: service.Name,
+							Port: test.HTTPBinPort,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "httpbin-facade-png",
+						Annotations: map[string]string{
+							// We'll use this annotation to test that modifications are allowed on KongServiceFacade
+							// level. This should make Kong Service created from this KongServiceFacade to always
+							// return a PNG image.
+							annotations.AnnotationPrefix + annotations.PathKey: "/image/png",
+							annotations.IngressClassKey:                        ingressClass,
+						},
+					},
+					Spec: incubatorv1alpha1.KongServiceFacadeSpec{
+						Backend: incubatorv1alpha1.KongServiceFacadeBackend{
+							Name: service.Name,
+							Port: test.HTTPBinPort,
+						},
+					},
+				},
+			}
+			clients := GetFromCtxForT[*clientset.Clientset](ctx, t)
+			for _, serviceFacade := range serviceFacades {
+				serviceFacade, err = clients.IncubatorV1alpha1().KongServiceFacades(namespace).Create(ctx, serviceFacade, metav1.CreateOptions{})
+				require.NoError(t, err)
+				cleaner.Add(serviceFacade)
+			}
+
+			t.Log("configuring Ingress")
+			ingress := builder.NewIngress(uuid.NewString(), ingressClass).WithRules(
+				netv1.IngressRule{
+					IngressRuleValue: netv1.IngressRuleValue{
+						HTTP: &netv1.HTTPIngressRuleValue{
+							Paths: []netv1.HTTPIngressPath{
+								{
+									Path:     "/jpeg",
```

Comment:
nit: this `/jpeg` and `/png` can be defined on top of this function as constants and used here and in assertion instead of verbatim strings